### PR TITLE
- Update man pages

### PR DESCRIPTION
**Pull request by rjschwei**
+ Provide man pages as target for implementation.

### CODE REVIEW

File: man/man1/mash-account-add-azure.1 [hunk 1]
```diff
@@ -1,12 +1,47 @@
...
-Add Azure account given the provided args.
+Add an Azure account in the user name space on the \fImash\fP server. See
+\fBmash-account-add(1)\fP for details about user name-spaces. Unless
+otherwise noted options are required.
 .SH OPTIONS
 .TP
-\fB\-\-group\fP TEXT
-Group name to place account in.
\ No newline at end of file
+\fB--name\fP ACCOUNT_NAME
```
Comment (smarlowucf) [inline man/man1/mash-account-add-azure.1]: If something is required why is an option? Options are normally optional. These are all really required arguments.

File: man/man1/mash-job-add.1 [hunk 2]
```diff
@@ -1,8 +1,92 @@
...
+is mutually exclusive with the \fIPARTITION\fP argument.
+.TP
+\fB--release\fP RELEASE
+Release the image. Mutually exclusive with the \fITEST\fP option
+.TP
+\fB--test\fP TEST
+Run the job through the testing stage of the pipeline. Mutually exclusive
+with the \fIRELEASE\fP option
+.TP
+\fB--tests-to-run\fP TESTS_TO_RUN
```
Comment (smarlowucf) [inline man/man1/mash-job-add.1]: Why not stay consistent in naming convention?

File: man/man1/mash-job-add.1 [hunk 3]
```diff
@@ -1,8 +1,92 @@
...
+The names of the tests to run during the test phase of the pipeline. The
+tests must be known to \fIIPA\fP.
+.TP
+\fB--schedule\fB TIME_AS_YYYYMMDD:HH:MM:SS
```
Comment (smarlowucf) [inline man/man1/mash-job-add.1]: Ditto.

File: man/man1/mash-job-add.1 [hunk 4]
```diff
@@ -1,8 +1,92 @@
...
+.TP
+\fB--old-image-name\fP OLD_IMAGE_NAME
+The name of the image in the cloud framework being replaced by this new
+image.
+.TP
+\fB--partition\fP PARTITION_NAME
+The name of the partition in which this image will be uploaded for testing
+or release. The image will be replicated to all regions in a partition.
+.TP
+\fB--regions\fP REGION_NAMES
```
Comment (smarlowucf) [inline man/man1/mash-job-add.1]: Wouldn't work this way, see above.

File: man/man1/mash-job-add.1 [hunk 5]
```diff
@@ -1,8 +1,92 @@
...
 .B mash job add
-[OPTIONS] DOCUMENT
+[OPTIONS]
 .SH DESCRIPTION
-Send add job request to mash server based on provided json document.
\ No newline at end of file
+Create a job on the \fImash\fP server for execution in the pipeline.
+.SH OPTIONS
+.TP
+\fB--account\fP
```
Comment (smarlowucf) [inline man/man1/mash-job-add.1]: There is the option to pass in a list of groups and a list of accounts. With the accounts option the parameters are a dictionary and in each framework that dictionary is different.

For example you can pass in a list of ec2 accounts and for each account you can have a list of regions.

The complexity of options is not accounted for here with `--account` and `--regions`.

**Comment by smarlowucf** [comment]
> Many command line tools have gotten away from the idea of forcing the user to remember positional command line arguments.

Fair. It is hard to format a command with the proper arguments in position if there are many. My concern is in the code where you end up with a ton of if/else clauses to manually handle requirements. However it looks like this can be handled by click with a `required=True` value.

File: man/man1/mash-job-add.1 (hunk 5 above)
Comment (rjschwei) [reply man/man1/mash-job-add.1]: The complexity is hidden in the account setup. Note how this says "--account" and not "--accounts" the user gets to specify only one name. Then "--regions" is expected to match the regions that match the partitions in this account.

File: man/man1/mash-job-add.1 (hunk 4 above)
Comment (rjschwei) [reply man/man1/mash-job-add.1]: Yes it does, works for all the ec2imgutils mash should not be any different.

File: man/man1/mash-account-add-azure.1 [hunk 6]
```diff
@@ -1,12 +1,47 @@
...
+The name of the resource group in the given \fIREGION\fP that holds the
+\fISTORAGE_ACCOUNT\fP that contains the given \fICONTAINER\fP.
+.TP
+\fB--storage-account\fP STORAGE_ACCOUNT_NAME
+The name of the storage account that contains the given
+\fICONTAINER\fP to which the image to be tested or released is to be
+uploaded.
+.TP
+\fB--credentials\fP CREDENTIALS_PATH
+??? Shouldn't we take the account key instead, or do we need the publish
```
Comment (smarlowucf) [inline man/man1/mash-account-add-azure.1]: Sorry didn't even see this question the first few times I reviewed this PR...

The credentials for Azure are a very specifically formatted json file. MASH and IPA used file based authentication that requires this format. https://ipa.readthedocs.io/en/latest/start.html#azure

File: man/man1/mash-job-add.1 [hunk 7]
```diff
@@ -1,8 +1,92 @@
...
+\fB--no-delete\fP NO_DELETE
+When a job is submitted for testing purposes the default is to delete
+the image form the framework if the tests pass. This option preserves
+the image in the framework.
+.TP
+\fB--old-image-name\fP OLD_IMAGE_NAME
+The name of the image in the cloud framework being replaced by this new
+image.
+.TP
+\fB--partition\fP PARTITION_NAME
```
Comment (smarlowucf) [inline man/man1/mash-job-add.1]: Should not be option here. There's no way to override an accounts partition. This is already set when an account is created.

File: man/man1/mash-job-add.1 (hunk 4 above)
Comment (smarlowucf) [reply man/man1/mash-job-add.1]: The original comment is in relation to the confusion over complexity. Given the discussion of the context I see this works.

However, MASH is for all providers not just EC2. --regions doesn't make sense for GCE or Azure. For those the argument is region. Although for GCE we really need a specific zone.

File: man/man1/mash-job-add.1 [hunk 8]
```diff
@@ -1,8 +1,92 @@
...
+\fB--allow-copy\fP ALLOW_COPY
+Flag to enable the account(s) with which the image is shared to copy the image
+by allowing copy access to the underly The option only applies
+when the \fIFRAMEWORK\fP is set to \fiec2\fP.
+.TP
+\fB--distro\fP DISTRO
+A short name for the distribution. Supported values
+\fIopensuse\P and \fIsles\fP
+.TP
+\fB--framework\fP FRAMEWORK
```
[the remark this answers is not in the document]
Comment (rjschwei) [reply man/man1/mash-job-add.1]: See comment in #6 we will need to use one of "--framework", "--platform", or "--cloud".

Provider is simply not the correct terminology. The provider is AWS, but we expect a value of "ec2". Also for OpenStack, if/when we need to expand there are going to be multiple providers. I know this change causes pain in multiple places and I apologize for being sloppy and not catching this earlier.

**Comment by rjschwei** [comment]
Going to keep this open as our rough guide. wording will eventually move into code or some other place to support man page generation.

**Comment by smarlowucf** [comment]
Closing this as the suggestions for man pages are included in the current version.

**Pull request closed without merging**